llvm/lib/CodeGen/SelectOptimize.cpp

```diff
@@ -763,11 +772,19 @@ void SelectOptimizeImpl::collectSelectGroups(BasicBlock &BB,
   };
 
   DenseMap<Value *, SelectLikeInfo> SelectInfo;
+  // Keeps visited comparisons to help identify AShr/LShr variants of auxiliary
+  // instructions.
+  SmallPtrSet<CmpInst *, 2> SeenCmp;
```

Comment:
It might be worth making this a SetVector to make sure the order is deterministic.